documentation/src/docs/watermarking.md

```diff
@@ -0,0 +1,121 @@
+# Watermarking
+
+
+
+
+Watermarking establishes points in time (watermarks) where Pravega asserts that all earlier events are written and read. All events earlier than the watermark can be processed, acted upon, or committed. The application does not need to wait for any more earlier events for the watermarked window of time.  
+
+In real-time stream processing, ingress is typically handled by multiple Pravega writers and reading is typically handled by mulitple Pravega readers. In this heavily parallel processing scenario, the Pravega controllers coordinate the positions of multiple writers and readers in the stream. Even so, events can ingress out of order and be read out of order, making it difficult for the application to know when a complete set of events exists for real-time calculations, decisionmaking, or transactional actions. How long should an application wait before deciding that it has all of the events for a defined time window?  Watermarking satisfies that question.
+
+Watermarking depends on optional features in Pravega writers and readers.
+* Pravega Writers must note time on each event (or use an optional feature that automatically timestamps events).
+* Pravega Readers request a time window. The lower bound of that window is the watermark. The application can then act on or process events before the watermark, knowing that the data is complete up to the last watermark.   
+
+Note: The watermark is a best ability mark. There is always the possibility that a reader is unusually delayed and recovers and processes an event that falls earlier than a watermark (earlier than a closed window of time). In such a case, it is up to the application to determine a recourse.
+
+Uses of watermarking include applications that require a time bounded   set of metrics to make a calculation or make a decision. Transactional applications are sometimes time-bound.
+
+The Flink API offers watermarking features. The Pravega watermarking feature mirrors the Flink functionality,   enabling an easier coding experience for integrating Pravega and Flink applications.
+
+## More information
+
+Read more about the Flink watermarking feature and example use cases <here>.
+Read more about the Pravega watermarking design and usage in the Pravega blog <here>.
+
+## About time
+
+Watermarking is an optional feature in Pravega. If implemented, the concept of time is defined by the application. The APIs provide  the means for writers to note the time  and for readers to request the watermarked window of time, but the definition of  time itself is  controlled by the application.
```

Comment:
In addition, please watch your spacing. There are multiple instances in this doc where you have 2 or more consecutive spaces.

Reply:
Resolved with latest rewrite